src/jit/hwintrinsiccodegenxarch.cpp

```diff
@@ -93,17 +93,34 @@ void CodeGen::genHWIntrinsic(GenTreeHWIntrinsic* node)
 
             case 2:
                 genConsumeOperands(node);
+                op1Reg = op1->gtRegNum;
+                op2Reg = op2->gtRegNum;
                 if (category == HW_Category_MemoryStore)
                 {
-                    emit->emitIns_AR_R(ins, emitTypeSize(TYP_SIMD16), op2->gtRegNum, op1->gtRegNum, 0);
+                    emit->emitIns_AR_R(ins, emitTypeSize(TYP_SIMD16), op2Reg, op1Reg, 0);
                 }
                 else if ((ival != -1) && varTypeIsFloating(baseType))
                 {
                     genHWIntrinsic_R_R_RM_I(node, ins);
                 }
                 else if (category == HW_Category_MemoryLoad)
                 {
-                    emit->emitIns_SIMD_R_R_AR(ins, emitTypeSize(TYP_SIMD16), targetReg, op1->gtRegNum, op2->gtRegNum);
+                    emit->emitIns_SIMD_R_R_AR(ins, emitTypeSize(TYP_SIMD16), targetReg, op1Reg, op2Reg);
+                }
+                else if (Compiler::isImmHWIntrinsic(intrinsicID, op2))
+                {
+                    if (op2->IsCnsIntOrI())
+                    {
+                        ssize_t ival = op2->AsIntConCommon()->IconValue();
```

Comment:
Since you have used `IsCnsIntOrI` using `AsIntCon` would make more sense. `AsIntConCommon` also accepts `GT_CNS_LNG` which is not needed here.

Reply:
Good point, thank you!